### PR TITLE
LPS-149303 Update calendar booking iterator

### PR DESCRIPTION
**Pull request by seltong**

### CODE REVIEW

**Comment by seltong** [comment]
ci:test:calendar

File: modules/apps/calendar/calendar-api/src/main/java/com/liferay/calendar/util/CalendarBookingIterator.java [hunk 1]
```diff
@@ -147,13 +151,23 @@ private boolean _isExceededCount() {
...
+	private DateValue _toDateValue(long time) throws ParseException {
+		SimpleDateFormat simpleDateFormat = new SimpleDateFormat(
+			DateUtil.ISO_8601_PATTERN);
+
+		simpleDateFormat.setTimeZone(_displayTimeZone);
+
+		String dateString = simpleDateFormat.format(new Date(time));
+
+		Date date = DateUtil.parseDate(
+			DateUtil.ISO_8601_PATTERN, dateString, LocaleUtil.getDefault());
```
Comment (MarinhoFeliphe) [inline modules/apps/calendar/calendar-api/src/main/java/com/liferay/calendar/util/CalendarBookingIterator.java:163]: Hi @seltong, declare `dateString` inline, please

Thanks!

File: modules/apps/calendar/calendar-api/src/main/java/com/liferay/calendar/util/CalendarBookingIterator.java [hunk 2]
```diff
@@ -147,13 +151,23 @@ private boolean _isExceededCount() {
...
+
+		Calendar calendar = JCalendarUtil.getJCalendar(date.getTime());
```
Comment (MarinhoFeliphe) [inline modules/apps/calendar/calendar-api/src/main/java/com/liferay/calendar/util/CalendarBookingIterator.java:165]: rename this variable to `jCalendar`, please

File: modules/apps/calendar/calendar-api/src/main/java/com/liferay/calendar/util/CalendarBookingIterator.java [hunk 3]
```diff
@@ -147,13 +151,23 @@ private boolean _isExceededCount() {
...
 
-		return new DateValueImpl(
-			jCalendar.get(Calendar.YEAR), jCalendar.get(Calendar.MONTH) + 1,
-			jCalendar.get(Calendar.DAY_OF_MONTH));
+		return new DateTimeValueImpl(
+			calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH),
```
Comment (MarinhoFeliphe) [inline modules/apps/calendar/calendar-api/src/main/java/com/liferay/calendar/util/CalendarBookingIterator.java:168]: Why this `calendar.get(Calendar.MONTH) + 1` was replaced by this `calendar.get(Calendar.MONTH)`?

Wouldn't we need to increment as well?

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by seltong** [comment]
ci:test:calendar

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:test:sf

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
Hi @seltong, could you plese check if these fails are related?

Thanks!

**Comment by seltong** [comment]
ci:forward

**Comment by seltong** [comment]
ci:test:relevant

**Comment by seltong** [comment]
ci:test:relevant

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:calendar

**Pull request closed without merging**